### PR TITLE
Refactored: NumberOf(Min,Max)PositionsCriterion

### PR DESCRIPTION
**Pull request by nimo23**
Fixes #.

Changes proposed in this pull request:
﻿- **Breaking:** **`NumberOfTradesCriterion`** renamed to **`NumberOfMaxPositionsCriterion`**
- :tada: **Enhancement** **`NumberOfMinPositionsCriterion`** 

- [x] added an entry with related ticket number(s) to the unreleased section of `CHANGES.md` 

Currently the `betterThan()` with NumberOfPositionsCriterion is vague. Thus, split this criterion to two and let the user decide if one is better than the other. For example, for backtesting:

- NumberOf**Min**PositionsCriterion: the more positions, the more tests were done, the better.
- NumberOf**Max**PositionsCriterion: the less positions, the better.

### CODE REVIEW

**Comment by team172011** [comment]
Whats the sense behind splitting this up into Min and Max? This criterion was just about counting Positions in a TradingRecord

File: ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/NumberOfMaxPositionsCriterion.java [hunk 1]
```diff
@@ -29,9 +29,9 @@
 import org.ta4j.core.num.Num;
 
 /**
- * Number of position criterion.
+ * Number of maxium allowed position criterion.
  */
-public class NumberOfPositionsCriterion extends AbstractAnalysisCriterion {
+public class NumberOfMaxPositionsCriterion extends AbstractAnalysisCriterion {
```
Comment (team172011) [inline ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/NumberOfMaxPositionsCriterion.java:34]: Why renaming this?

File: ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/NumberOfMinPositionsCriterion.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * Number of minimum required position criterion.
+ */
+public class NumberOfMinPositionsCriterion extends AbstractAnalysisCriterion {
+
+    @Override
+    public Num calculate(BarSeries series, TradingRecord tradingRecord) {
+        return series.numOf(tradingRecord.getPositionCount());
+    }
```
Comment (team172011) [inline ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/NumberOfMinPositionsCriterion.java:39]: Same as in NumberOfPositionsCriterion that has been renamed.

File: ta4j-core/src/test/java/org/ta4j/core/analysis/criteria/NumberOfMinPositionsCriterionTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    public NumberOfMinPositionsCriterionTest(Function<Number, Num> numFunction) {
+        super((params) -> new NumberOfMinPositionsCriterion(), numFunction);
+    }
+
+    @Test
+    public void calculateWithNoPositions() {
+        MockBarSeries series = new MockBarSeries(numFunction, 100, 105, 110, 100, 95, 105);
+
+        AnalysisCriterion buyAndHold = getCriterion();
+        assertNumEquals(0, buyAndHold.calculate(series, new BaseTradingRecord()));
```
Comment (team172011) [inline ta4j-core/src/test/java/org/ta4j/core/analysis/criteria/NumberOfMinPositionsCriterionTest.java:52]: ?

**Review by team172011 — CHANGES_REQUESTED** [review]

File: ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/NumberOfMaxPositionsCriterion.java (hunk 1 above)
Comment (nimo23) [reply ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/NumberOfMaxPositionsCriterion.java:34]: Currently the betterThan() with NumberOfPositionsCriterion is vague. Look at the current impl of `betterThan()`. **Why is having more positions better?** Thus, split this criterion to two and let the user decide if one is better than the other. For example, for backtesting:

    NumberOfMinPositionsCriterion: the more positions, the more tests were done, the better.
    NumberOfMaxPositionsCriterion: the less positions, the better.

That's the idea why I wanted to split it to two criterions. Should I add new criterions instead? `NumberOfMinMaxPositionsCriterion` and leave the current as it is?

**Comment by nimo23** [comment]
>Whats the sense behind splitting this up into Min and Max? This criterion was just about counting Positions in a TradingRecord

@team172011 But criterions are ment to be comparable by `betterThan()`. With the actual implementation, it is supposed that _the more trades the better_. But this is not always the case and depend on the users scenario. Hence, split this indicator:

- `NumberOfMinPositionsCriterion`:  the more trades, the better.
- `NumberOfMaxPositionsCriterion`: the less trades, the better.

**Comment by team172011** [comment]
Okay I unterstand, but than I would prefer to delete this criterions. In both ways it is a heavily overengineered feature for just counting the trades

**Pull request closed without merging**